tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,229 @@
+    assert comment_4['is_moderator'] is False
+    # Because the serializer does not differentiate between
+    # moderator and creator for deleting the comment,
+    # does it actually make sense to show this
+    # differentiation in the serializer?
+
+    # Why is there an is_removed property
+    # if there is also an is_deleted property?
+
+    # Testing a comment that has been marked by a moderator
+    comment_6 = comment_data[1]
+    assert comment_6['id'] == 6
+    assert comment_6['user_name'] == ''
+    assert comment_6['user_pk'] == str(user.pk)
+    assert comment_6['user_profile_url'] == ''
+    assert comment_6['is_deleted'] is False
+    assert comment_6['is_moderator'] is False
+    assert len(comment_6['child_comments']) == 0
+    assert comment_6['comment'] == Comment.objects.get(pk=6).comment
+    assert comment_6['object_pk'] == question.pk
+    assert comment_6['is_removed'] is False
+    assert comment_6['is_censored'] is False
+    assert comment_6['is_blocked'] is False
+    assert len(comment_6['comment_categories']) == 0
+    assert comment_6['last_discussed'] is None
+    assert comment_6['is_moderator_marked'] is True
+    assert comment_6['content_type'] == question_ct.pk
+    # Moderator-marked comments no longer show the user name
+    # nor the link to the user profile. Does this make sense?
+
+    # Testing a comment with child-comments:
+    comment_7 = comment_data[0]
+    assert comment_7['id'] == 7
+    assert comment_7['user_pk'] == str(moderator.pk)
+    assert comment_7['is_deleted'] is False
+    assert comment_7['is_moderator'] is True
+    assert len(comment_7['child_comments']) == 1
+    assert comment_7['comment'] == Comment.objects.get(pk=7).comment
+    assert comment_7['object_pk'] == question.pk
+    assert comment_7['is_removed'] is False
+    assert comment_7['is_censored'] is False
+    assert comment_7['is_blocked'] is False
+    assert len(comment_7['comment_categories']) == 0
+    assert comment_7['last_discussed'] is None
+    assert comment_7['is_moderator_marked'] is False
+    assert comment_7['content_type'] == question_ct.pk
+
+    # Testing a child-comment:
+    comment_8 = comment_7['child_comments'][0]
+    assert comment_8['id'] == 8
+    assert comment_8['user_pk'] == str(user.pk)
+    assert comment_8['is_deleted'] is False
+    assert comment_8['is_moderator'] is False
+    assert comment_8['comment'] == Comment.objects.get(pk=8).comment
+    assert comment_8['object_pk'] == comment.pk
+    assert comment_8['is_removed'] is False
+    assert comment_8['is_censored'] is False
+    assert comment_8['is_blocked'] is False
+    assert len(comment_8['comment_categories']) == 0
+    assert comment_8['last_discussed'] is None
+    assert comment_8['is_moderator_marked'] is False
+    assert comment_8['content_type'] == comment_ct.pk
+
+    # Notes:
+    # The property 'last_discussed' is tested in test_signals.py.
+    # Is the property 'is_moderator_marked' ever actually used?
```

Comment:
yes, for an ordering filter in one project. We can just keep it (and either remove it at some point when we do not need it anymore or also use it in another project.